src/fields/BooleanField.h

```diff
@@ -1,22 +1,24 @@
-#ifndef BOOLEANFIELD_H
-#define BOOLEANFIELD_H
-
-#include "Field.h"
-
-using namespace std;
-
-class BooleanField: public Field {
-
-    private:
-        bool _fieldValue;
-
-    public:
-        BooleanField(string fieldName, bool defaultValue=false, bool isPrimaryKey=false, bool isUnique=false, bool isNullable=true);
-
-        void setValue(bool value);
-        bool getValue();
-
-        void setParameter(int parameterNumber, DBConnection connection);
-};
-
-#endif
+#ifndef BOOLEANFIELD_H
+#define BOOLEANFIELD_H
+
+#include "Field.h"
+
+using namespace std;
+
+class BooleanField: public Field {
+
+    private:
+        bool _fieldValue;
+        bool _defaultValue;
```

Comment:
Exactly. But I am still wondering if `_isAutoFilled` is needed since setting it to `false` would be the same as setting `_defaultValue` to `false`.